packages/core/test/SchemaField_test.js

```diff
@@ -292,6 +292,50 @@ describe("SchemaField", () => {
     });
   });
 
+  describe("additional properties", () => {
+    const schema = {
+      type: "object",
+      additionalProperties: true,
+    };
+
+    it("should render text input for object key if schema has additional properties", () => {
+      const { node } = createFormComponent({ schema });
+      Simulate.click(node.querySelector(".glyphicon-plus"));
+
+      expect(node.querySelectorAll("input[value='newKey']")).to.have.length.of(
+        1
+      );
+    });
+
+    it("should render SchemaField as input for object key if schema has propertyNames schema", () => {
+      const schemaWithPropertyNames = {
+        ...schema,
+        propertyNames: {
+          type: "string",
+          enum: ["1", "2"],
+        },
+      };
+      const { node } = createFormComponent({ schema: schemaWithPropertyNames });
+      Simulate.click(node.querySelector(".glyphicon-plus"));
+
+      expect(node.querySelectorAll("select")).to.have.length.of(1);
+    });
+
+    it("should render SchemaField as input for object key if schema has propertyNames schema without type", () => {
+      const schemaWithPropertyNames = {
+        ...schema,
+        propertyNames: {
+          enum: ["1", "2"],
+          pattern: "^[A-Za-z_][A-Za-z0-9_]*$",
+        },
+      };
+      const { node } = createFormComponent({ schema: schemaWithPropertyNames });
+      Simulate.click(node.querySelector(".glyphicon-plus"));
+
+      expect(node.querySelectorAll("select")).to.have.length.of(1);
+    });
```

Comment:
Let's add more tests for robustness:

```
{propertyNames: {oneOf: [{const: "a"}, const: "b"}]}}

{propertyNames: {$ref: "#/definitions/test"}}

{propertyNames: {type: "boolean"}}

{propertyNames: {const: "test"}}
```

and make sure the appropriate widget renders properly.